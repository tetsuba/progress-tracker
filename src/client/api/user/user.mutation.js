import gql from 'graphql-tag'

export const VERIFY_EMAIL_MUTATION = gql`
  mutation($input: UserEmailInput!) {
    verifyEmail(input: $input) {
      confirmation
    }
  }
`

export const SEND_PASSWORD_RESET_CONFIRMATION_MUTATION = gql`
  mutation($input: UserEmailInput!) {
    sendPasswordResetConfirmation(input: $input) {
      confirmation
    }
  }
`

export const LOGIN_MUTATION = gql`
  mutation($input: LoginUserInput!) {
    userLogin(input: $input) {
      id
      firstName
      lastName
      token
      email
    }
  }
`

export const REGISTER_USER_MUTATION = gql`
  mutation NewUser($input: NewUserInput!) {
    newUser(input: $input) {
      id
    }
  }
`

export const REST_PASSWORD_MUTATION = gql`
  mutation ResetPassword($input: ResetPasswordInput!) {
    resetPassword(input: $input) {
      confirmation
    }
  }
`
