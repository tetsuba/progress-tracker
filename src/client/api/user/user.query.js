import gql from 'graphql-tag'

export const GET_USER_DETAILS_QUERY = gql`
  {
    getUserDetails {
      firstName
      lastName
      email
    }
  }
`

export const IS_USER_SESSION_EXPIRED = gql`
  {
    isUserSessionExpired {
      success
    }
  }
`

export const VALIDATE_USER_EMAIL_QUERY = gql`
  query ValidateUserEmail($token: String!) {
    validateUserEmail(token: $token) {
      success
    }
  }
`
