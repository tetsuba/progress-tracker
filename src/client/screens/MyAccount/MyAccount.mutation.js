import gql from 'graphql-tag';

export const UPDATE_USER_MUTATION = gql`    
    mutation($input: UserDataInput!) {
        updateUserData(input: $input) {
            firstName
            lastName
            id
        }
    }
`
