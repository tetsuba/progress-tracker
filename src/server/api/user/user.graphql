
type UserType {
    id: String
    firstName: String
    lastName: String
    email: String
    password: String
    country: String
}

type UserLoginType {
    id: String
    firstName: String
    lastName: String
    token: String
    email: String
}

type UserVerifyType {
    confirmation: String
}

input NewUserInput {
    firstName: String
    lastName: String
    email: String
    password: String
    country: String
}

input LoginUserInput {
    email: String
    password: String
}

input UserEmailInput {
    email: String
}

input UserDataInput {
    id: String
    firstName: String
    lastName: String
}

input ResetPasswordInput {
    id: String
    password: String
}

extend type Query {
    getUserData: UserType!
    isUserSessionExpired(token: String): UserType!
}

type Mutation {
    newUser(input: NewUserInput!): UserType!
    userLogin(input: LoginUserInput!): UserLoginType!
    updateUserData(input: UserDataInput!): UserLoginType!
    verifyEmail(input: UserEmailInput!): UserVerifyType!
    sendPasswordResetConfirmation(input: UserEmailInput!): UserVerifyType!
    resetPassword(input: ResetPasswordInput!): UserVerifyType!
}
