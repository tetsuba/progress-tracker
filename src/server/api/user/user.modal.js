const mongoose = require('mongoose')

const userSchema = new mongoose.Schema(
  {
    firstName: {
      type: String,
      required: true,
    },
    lastName: {
      type: String,
      required: true,
    },
    email: {
      type: String,
      required: true,
    },
    password: {
      type: String,
      required: true,
    },
    country: {
      type: String,
      required: false,
    },
    isVerified: {
      type: Boolean,
      default: false,
    },
  },
  { _id: true }
)

module.exports = userSchema
