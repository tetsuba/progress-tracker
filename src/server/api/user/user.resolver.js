const { validateEmail } = require('../../utils/common')

const {
  createNewUser,
  findUser,
  updateUser,
  findEmail,
  resetPassword,
} = require('./user.CRUD')
const { getEmailMailOptions } = require('../../utils/email/sendEmail.js')
const { errorName } = require('../../errorHandling')
const { createVerificationToken } = require('../../utils/token')
const { passwordsDoNotMatch } = require('../utils')

module.exports = {
  Query: {
    isUserSessionExpired: async (_, args, context) => {
      try {
        return await findUser({
          email: context.user.email,
          password: context.user.password,
        })
      } catch (e) {
        return e
      }
    },
    getUserData: async (_, args, context) => {
      console.log('@@@@@@@@@@@@ getUserData: ', context)
      try {
        return await findUser({
          email: context.user.email,
          password: context.user.password,
        })
      } catch (e) {
        return e
      }
    },
  },

  Mutation: {
    newUser: async (_, args, context, info) => {
      const name = 'confirmEmail'

      if (validateEmail(args.input.email)) {
        throw new Error(errorName.NOT_VALID_EMAIL)
      }

      try {
        const mailOptions = getEmailMailOptions(name)
        return await createNewUser(args.input, mailOptions)
      } catch (err) {
        return err
      }
    },

    userLogin: async (_, args, context, info) => {
      const { User, Token } = context.models
      const {
        input: { email, password },
      } = args
      const user = await User.findOne({ email })

      /* Error Handling:
       * - User is not found
       * - Passwords do not match
       *  */
      if (!user || passwordsDoNotMatch(password, user.password)) {
        throw new Error(errorName.INCORRECT_USER_DETAILS)
      }

      /* Error Handling:
       * - User email has not been verified
       *  */
      if (!user.isVerified) {
        throw new Error(errorName.EMAIL_NOT_VERIFIED)
      }

      /* Create a session token (12hrs):
       *  */
      const verificationToken = createVerificationToken(user)
      const { token } = await Token.create({
        _userId: user._id,
        token: verificationToken,
      })

      return { token }
    },

    updateUserData: async (_, args, context, info) => {
      try {
        return await updateUser(args.input)
      } catch (err) {
        return err
      }
    },

    verifyEmail: async (_, args, context) => {
      const name = 'confirmEmail'
      try {
        const mailOptions = getEmailMailOptions(name)
        return await findEmail({ email: args.input.email }, mailOptions)
      } catch (err) {
        if (err.name === 'TypeError') {
          console.log(
            `Property "name": "${name}" is incorrect and can not get mail options`
          )
          return new Error(errorName.INTERNAL_SERVER_ERROR)
        }
        return err
      }
    },

    sendPasswordResetConfirmation: async (_, args, context) => {
      const name = 'resetPassword'
      try {
        const mailOptions = getEmailMailOptions(name)
        return await findEmail({ email: args.input.email }, mailOptions)
      } catch (err) {
        if (err.name === 'TypeError') {
          console.log(
            `Property "name": "${name}" is incorrect and can not get mail options`
          )
          return new Error(errorName.INTERNAL_SERVER_ERROR)
        }
        return err
      }
    },

    resetPassword: async (_, args, context) => {
      try {
        return await resetPassword(args.input.token, args.input.password)
      } catch (err) {
        return err
      }
    },
  },
}

// NOTES
// fieldName: (parent, args, context, info) => data;

// parent:  An object that contains the result returned from the resolver on the parent type
// args:    An object that contains the arguments passed to the field
// context: An object shared by all resolvers in a GraphQL operation. We use the context to contain per-request state such as authentication information and access our data sources.
// info:    Information about the execution state of the operation which should only be used in advanced cases
