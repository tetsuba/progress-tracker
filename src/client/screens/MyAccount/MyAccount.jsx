import React from 'react'
import { useMutation, useQuery } from '@apollo/react-hooks'

// COMPONENTS
import { Container, Row } from 'react-bootstrap'
import BreadCrumbs from '../../components/BreadCrumbs/BreadCrumbs'

// FORM
import MyDetailsForm from '../../components/Form/MyDetailsForm'

// QUERY
import { GET_USER_DETAILS_QUERY } from '../../api/user/user.query'

// MUTATION
import { UPDATE_USER_DETAILS_MUTATION } from '../../api/user/user.mutation'

// COMPONENTS
import Box from '../../components/Box/Box'
import ForgotMyPasswordForm from '../../components/Form/ForgotMyPasswordForm'

// BREADCRUMBS
const crumbs = [
  { path: '/', name: 'Home' },
  { path: '', name: 'My Account' },
]

export default function MyAccount() {
  const { loading, data } = useQuery(GET_USER_DETAILS_QUERY)
  const [updateUserDetails, updateUserDetailsOptions] = useMutation(
    UPDATE_USER_DETAILS_MUTATION,
    {
      refetchQueries: [{ query: GET_USER_DETAILS_QUERY }],
    }
  )

  if (loading) {
    return <div>Page Loader</div>
  }

  return (
    <Container className="pt-5">
      <BreadCrumbs crumbs={crumbs} />
      <MyDetailsForm
        handleSubmit={(options, setShowFormElement) => {
          updateUserDetails(options)
            .then(() => setShowFormElement(false))
            .catch(() => console.log('Error!'))
        }}
        error={updateUserDetailsOptions.error}
        user={data.getUserDetails}
      />

      <Row className="mt-5">
        <ForgotMyPasswordForm
          hideEmailInput
          defualtEmail={data.getUserDetails.email}
          buttonText="Send"
        >
          <h3>Request for a password reset</h3>
          <p>We will send you a link to reset your password.</p>
        </ForgotMyPasswordForm>
      </Row>

      <Row className="mt-5">
        <Box max={500}>
          <h3>Email - Changing email address is not available</h3>
          <p>{data.getUserDetails.email}</p>
        </Box>
      </Row>
    </Container>
  )
}
