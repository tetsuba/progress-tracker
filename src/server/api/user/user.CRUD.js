const { getUserId, deleteToken } = require('../token/token.CRUD');
const jwt = require('jsonwebtoken');

// This order can not be changed
const mongoose = require('mongoose');
require('./user.modal');
const User = mongoose.model('user');
// -----------------------------

const { createToken } = require('../token/token.CRUD');
const { sendMail } = require('../../utils/email/sendEmail');
const { errorName } = require('../../errorHandling');
const { createAuthToken } = require('../../utils/token');

async function findEmail(data, mailOptions) {
    try {
        const user = await User.findOne({ email: data.email });

        // ERROR - User not found
        if (!user) {
            throw new Error(errorName.EMAIL_DOES_NOT_EXIST)
        }

        // ERROR - User is verified
        if (user.isVerified && mailOptions.subject.includes('confirm')) {
            throw new Error(errorName.EMAIL_VERIFIED);
        }

        // Do not send an email when running functional tests
        if (!process.env.TEST) {
            const obj = await createToken(user);
            await sendMail(obj.token, mailOptions);
        }

        return {
            confirmation: 'Please check your email and confirm.'
        }

    } catch(error) {
        return error
    }

}

async function findUser(data) {
    try {
        const user = await User.findOne({ email: data.email });

        // ERROR - User is not found or password does not match
        if (!user || user.password !== data.password) {
            throw new Error(errorName.INCORRECT_USER_DETAILS)
        }

        // ERROR - Email address not verified
        if (!user.isVerified) {
            throw new Error(errorName.EMAIL_NOT_VERIFIED)
        }

        const token = createAuthToken(user);

        const payload = jwt.verify(
            token,
            process.env.REACT_APP_JWT_AUTH_SECRET
    )

        return {
            firstName: user.firstName,
            lastName: user.lastName,
            email: user.email,
            id: user._id,
            token
        }
    } catch(error) {
        return error
    }
}

// RegisterUser
async function createNewUser(data) {
    try {
        const user = await User.findOne({ email: data.email});

        // ERROR - User is not found or password does not match
        if (user) {
            throw new Error(errorName.EMAIL_ALREADY_EXIST)
        }

        const newUser = await User.create(data);
        const obj = await createToken(newUser);
        sendMail(obj.token);
        return { success: 'success' };
    } catch(err) {
        return err
    }
}

async function updateUser(data) {
    console.log('updateUser:', data);
    try {
        const user = await User.findById(data.id);
        console.log('User:', user);
        user.firstName = data.firstName;
        user.lastName = data.lastName;
        return await user.save()
    } catch (e) {
        console.log('findUser error', e);
        return e;
    }
}

async function userVerified(id) {
    try {
        const user = await User.findById(id);
        user.isVerified = true;
        return await user.save()
    } catch (e) {
        console.log('userVerified error', e);
        return e;
    }
}

async function resetPassword(token, password) {
    try {
        const id = await getUserId(token);
        const user = await User.findById(id);
        user.password = password;
        await user.save();

        // Don not delete token when running functional tests
        if (!process.env.TEST) {
            await deleteToken(token);
        }

        return {
            confirmation: 'New password is saved'
        }
    } catch (e) {
        console.log('resetPassword error', e);
        return e;
    }
}

module.exports = {
    findUser,
    createNewUser,
    updateUser,
    userVerified,
    findEmail,
    resetPassword,
};
