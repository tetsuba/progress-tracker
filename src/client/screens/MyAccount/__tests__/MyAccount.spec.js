import { act } from 'react-dom/test-utils'
import { delay, graphRenderer } from '../../../../test/testHelper'
import MyAccount from '../../MyAccount/MyAccount'
import { getUserDetailsQuerySuccess } from '../../../../test/mockApi/user/userMockQuery'

jest.mock('react-router-dom', () => ({
  Link: ({ children }) => children,
}))

describe('<MyAccount>', () => {
  it('Should render', async () => {
    let wrapper
    await act(async () => {
      wrapper = graphRenderer(MyAccount, [getUserDetailsQuerySuccess], {})
      await delay()
    })
    wrapper.update()
    expect(wrapper.find(MyAccount)).toMatchSnapshot()
  })
})
