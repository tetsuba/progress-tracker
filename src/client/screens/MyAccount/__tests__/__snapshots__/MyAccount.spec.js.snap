// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyAccount> Should render 1`] = `
<MyAccount
  pageState="resetPassword"
>
  <Container
    className="pt-5"
    fluid={false}
  >
    <div
      className="pt-5 container"
    >
      <BreadCrumbs
        crumbKey="myAccount"
      >
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Breadcrumb
              label="breadcrumb"
              listProps={Object {}}
            >
              <nav
                aria-label="breadcrumb"
              >
                <ol
                  className="breadcrumb"
                >
                  <Crumbs>
                    <li
                      className="breadcrumb-item"
                      key="Home"
                    >
                      <Link
                        to="/"
                      >
                        Home
                      </Link>
                    </li>
                    <BreadcrumbItem
                      active={true}
                      key="My Account"
                      linkProps={Object {}}
                    >
                      <li
                        aria-current="page"
                        className="breadcrumb-item active"
                      >
                        My Account
                      </li>
                    </BreadcrumbItem>
                  </Crumbs>
                </ol>
              </nav>
            </Breadcrumb>
          </div>
        </Row>
      </BreadCrumbs>
      <MyDetailsForm
        user={
          Object {
            "email": "test@test.com",
            "firstName": "John",
            "lastName": "Doh",
          }
        }
      >
        <Row
          noGutters={false}
        >
          <div
            className="row"
          >
            <Box
              max={500}
            >
              <div
                style={
                  Object {
                    "backgroundColor": "#f8f9fa",
                    "border": "solid 1px #ccc",
                    "borderRadius": "8px",
                    "margin": "0 auto",
                    "maxWidth": 500,
                    "padding": "20px 20px",
                    "width": "100%",
                  }
                }
              >
                <h3
                  className="mb-5"
                >
                  My Details
                </h3>
                <Form
                  id="MyDetailsForm"
                  inline={false}
                  onSubmit={[Function]}
                >
                  <form
                    className=""
                    id="MyDetailsForm"
                    onSubmit={[Function]}
                  >
                    <FormGroup
                      className="pb-3"
                      controlId="formPlaintextFirstName"
                    >
                      <div
                        className="pb-3 form-group"
                      >
                        <FormLabel
                          className="text-muted"
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="text-muted form-label"
                            htmlFor="formPlaintextFirstName"
                          >
                            First Name:
                          </label>
                        </FormLabel>
                        <FormControl
                          isInvalid={false}
                          name="firstName"
                          onChange={[Function]}
                          placeholder="John"
                          readOnly={true}
                          required={true}
                          type="text"
                          value="John"
                        >
                          <input
                            className="form-control"
                            id="formPlaintextFirstName"
                            name="firstName"
                            onChange={[Function]}
                            placeholder="John"
                            readOnly={true}
                            required={true}
                            type="text"
                            value="John"
                          />
                        </FormControl>
                      </div>
                    </FormGroup>
                    <FormGroup
                      controlId="formPlaintextPassword"
                    >
                      <div
                        className="form-group"
                      >
                        <FormLabel
                          className="text-muted"
                          column={false}
                          srOnly={false}
                        >
                          <label
                            className="text-muted form-label"
                            htmlFor="formPlaintextPassword"
                          >
                            Last Name:
                          </label>
                        </FormLabel>
                        <FormControl
                          isInvalid={false}
                          name="lastName"
                          onChange={[Function]}
                          placeholder="Doh"
                          readOnly={true}
                          required={true}
                          type="input"
                          value="Doh"
                        >
                          <input
                            className="form-control"
                            id="formPlaintextPassword"
                            name="lastName"
                            onChange={[Function]}
                            placeholder="Doh"
                            readOnly={true}
                            required={true}
                            type="input"
                            value="Doh"
                          />
                        </FormControl>
                        <Feedback
                          type="invalid"
                        >
                          <div
                            className="invalid-feedback"
                          />
                        </Feedback>
                      </div>
                    </FormGroup>
                    <Button
                      active={false}
                      className="float-right"
                      disabled={false}
                      id="MyDetailsFormEdit"
                      onClick={[Function]}
                      style={
                        Object {
                          "width": 100,
                        }
                      }
                      type="button"
                      variant="primary"
                    >
                      <button
                        className="float-right btn btn-primary"
                        disabled={false}
                        id="MyDetailsFormEdit"
                        onClick={[Function]}
                        style={
                          Object {
                            "width": 100,
                          }
                        }
                        type="button"
                      >
                        Edit
                      </button>
                    </Button>
                  </form>
                </Form>
              </div>
            </Box>
          </div>
        </Row>
      </MyDetailsForm>
      <Row
        className="mt-5"
        noGutters={false}
      >
        <div
          className="mt-5 row"
        >
          <ForgotMyPasswordForm
            buttonText="Send"
            defaultEmail="test@test.com"
            hideEmailInput={true}
            setPageState={[Function]}
          >
            <Box
              max={500}
            >
              <div
                style={
                  Object {
                    "backgroundColor": "#f8f9fa",
                    "border": "solid 1px #ccc",
                    "borderRadius": "8px",
                    "margin": "0 auto",
                    "maxWidth": 500,
                    "padding": "20px 20px",
                    "width": "100%",
                  }
                }
              >
                <Row
                  noGutters={false}
                >
                  <div
                    className="row"
                  >
                    <Col>
                      <div
                        className="col"
                      >
                        <h3>
                          Request for a password reset
                        </h3>
                        <p>
                          We will send you a link to reset your password.
                        </p>
                        <Form
                          className="w-100"
                          id="ForgotMyPasswordForm"
                          inline={false}
                          onSubmit={[Function]}
                        >
                          <form
                            className="w-100"
                            id="ForgotMyPasswordForm"
                            onSubmit={[Function]}
                          >
                            <FormGroup
                              controlId="ForgotMyPasswordEmail"
                            >
                              <div
                                className="form-group"
                              >
                                <FormControl
                                  isInvalid={false}
                                  name="email"
                                  onChange={[Function]}
                                  placeholder="test@test.com"
                                  readOnly={true}
                                  required={true}
                                  type="email"
                                  value="test@test.com"
                                >
                                  <input
                                    className="form-control"
                                    id="ForgotMyPasswordEmail"
                                    name="email"
                                    onChange={[Function]}
                                    placeholder="test@test.com"
                                    readOnly={true}
                                    required={true}
                                    type="email"
                                    value="test@test.com"
                                  />
                                </FormControl>
                                <Feedback
                                  type="invalid"
                                >
                                  <div
                                    className="invalid-feedback"
                                  />
                                </Feedback>
                                <FormText
                                  className="text-muted"
                                >
                                  <small
                                    className="text-muted form-text"
                                  />
                                </FormText>
                              </div>
                            </FormGroup>
                            <Button
                              active={false}
                              className="float-right"
                              disabled={false}
                              id="ForgotMyPasswordSubmit"
                              type="submit"
                              variant="primary"
                            >
                              <button
                                className="float-right btn btn-primary"
                                disabled={false}
                                id="ForgotMyPasswordSubmit"
                                type="submit"
                              >
                                Reset Password
                              </button>
                            </Button>
                          </form>
                        </Form>
                        <TextLink
                          eventHandler={[Function]}
                        >
                          <span
                            id="TextLink"
                            onClick={[Function]}
                            style={
                              Object {
                                "color": "#007bff",
                                "cursor": "pointer",
                              }
                            }
                          >
                            <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon={
                                Object {
                                  "icon": Array [
                                    192,
                                    512,
                                    Array [],
                                    "f0d9",
                                    "M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z",
                                  ],
                                  "iconName": "caret-left",
                                  "prefix": "fas",
                                }
                              }
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              swapOpacity={false}
                              symbol={false}
                              title=""
                              transform={null}
                            >
                              <svg
                                aria-hidden="true"
                                className="svg-inline--fa fa-caret-left fa-w-6 "
                                data-icon="caret-left"
                                data-prefix="fas"
                                focusable="false"
                                role="img"
                                style={Object {}}
                                viewBox="0 0 192 512"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
                                  fill="currentColor"
                                  style={Object {}}
                                />
                              </svg>
                            </FontAwesomeIcon>
                             back
                          </span>
                        </TextLink>
                      </div>
                    </Col>
                  </div>
                </Row>
              </div>
            </Box>
          </ForgotMyPasswordForm>
        </div>
      </Row>
      <Row
        className="mt-5"
        noGutters={false}
      >
        <div
          className="mt-5 row"
        >
          <Box
            max={500}
          >
            <div
              style={
                Object {
                  "backgroundColor": "#f8f9fa",
                  "border": "solid 1px #ccc",
                  "borderRadius": "8px",
                  "margin": "0 auto",
                  "maxWidth": 500,
                  "padding": "20px 20px",
                  "width": "100%",
                }
              }
            >
              <h3>
                Email - Changing email address is not available
              </h3>
              <p>
                test@test.com
              </p>
            </div>
          </Box>
        </div>
      </Row>
    </div>
  </Container>
</MyAccount>
`;
